wms/models/datasets/ugrid.py

```diff
@@ -284,7 +281,29 @@ def getmap(self, layer, request):
                     return self.empty_response(layer, request)
 
                 if request.GET['image_type'] == 'filledcontours':
-                    return mpl_handler.tricontourf_response(tri_subset, data, request)
+                    if np.isnan(np.sum(data)):
+                        data_mask = np.logical_not(np.isnan(data))  # mask for non-NaN data elements
```

Comment:
I wonder is `~np.isnan(data)` is faster than `np.logical_not(np.isnan(data))`.

Anyways, by changing the statement above to:
```python
mask = np.isnan(data)
if mask.any():
.
.
.
 mask data
```

You'll can np.<something> only once instead of 4 times.